zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/GunStats.java

```diff
@@ -1,45 +1,60 @@
 package com.github.phantazmnetwork.zombies.equipment.gun;
 
-import com.github.phantazmnetwork.commons.Namespaces;
 import com.github.phantazmnetwork.zombies.equipment.gun.shoot.handler.ShotHandler;
+import com.github.steanky.element.core.annotation.DataObject;
+import com.github.steanky.element.core.annotation.FactoryMethod;
+import com.github.steanky.element.core.annotation.Model;
+import com.github.steanky.element.core.annotation.ProcessorMethod;
 import com.github.steanky.ethylene.core.ConfigElement;
 import com.github.steanky.ethylene.core.collection.ConfigNode;
 import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
 import com.github.steanky.ethylene.core.processor.ConfigProcessException;
 import com.github.steanky.ethylene.core.processor.ConfigProcessor;
-import net.kyori.adventure.key.Key;
-import net.kyori.adventure.key.Keyed;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Objects;
+
 /**
  * A gun's generic stats. These stats only pertain to shooting, not aspects such as damage or knockback.
  * These are handled by individual {@link ShotHandler}s.
- *
- * @param shootSpeed   The gun's shoot speed
- * @param reloadSpeed  The gun's reload speed
- * @param maxAmmo      The gun's max ammo
- * @param maxClip      The gun's max clip
- * @param shots        The gun's shots per clip
- * @param shotInterval The interval between gun fire
  */
-public record GunStats(long shootSpeed, long reloadSpeed, int maxAmmo, int maxClip, int shots, long shotInterval)
-        implements Keyed {
+@Model("zombies.gun.stats")
+public final class GunStats {
```

Comment:
this should probably be annotated with @Cache(false), that way new instances of GunStats will always be created

Reply:
why would we want that?

Reply:
if you make multiple separate GunLevel instances from the same config, you don't want them sharing GunStats

Reply:
doesnt that apply to everything

Reply:
when no @Cache is explicitly provided, it's up to the caller of ElementContext#provide whether the object gets cached or not
so adding @Cache(false) to objects is a safety measure

Reply:
but would it be best to apply it to everything?

Reply:
not everything, since for some objects it doesn't matter if they get cached or not

Reply:
example?

Reply:
any objects that don't have state can be cached, as long as they don't have child objects that *do* have state, like GunLevel

Reply:
GunStats don't have state?